internal/util/kubernetes/kubernetes.go

```diff
@@ -86,10 +90,44 @@ func GetSecret(ctx context.Context, client k8sClient.Client, namespace, secretNa
 	return secret, nil
 }
 
+func DeleteKubernetesResource(ctx context.Context, client k8sClient.Client, item k8sClient.Object) error {
+	if err := client.Delete(ctx, item); err != nil {
+		return fmt.Errorf("error deleting resource %s/%s: %v", item.GetNamespace(), item.GetName(), err)
+	}
+	return nil
+}
+
 func IsValidKubernetesManifestFile(fileName string) bool {
 	fileExt := strings.Split(fileName, ".")
 	if _, ok := validManifestExtensions[fileExt[len(fileExt)-1]]; ok {
 		return true
 	}
 	return false
 }
+
+// DeleteResourcesByAnnotations deletes all resources of a given GroupVersionKind across the kubernetes cluster
+func DeleteResourcesByAnnotations(ctx context.Context, client k8sClient.Client, gvks []schema.GroupVersionKind, annotationKey string, annotationValue string) error {
+	logger := logging.FromContext(ctx)
+	for _, gvk := range gvks {
+		listItems := &unstructured.UnstructuredList{}
+		listItems.SetGroupVersionKind(gvk)
+
+		// List all resources of the given GroupVersionKind across the cluster
+		if err := client.List(ctx, listItems); err != nil {
```

Comment:
this will eventually be a very large list of items, so a preferred approach would be to use a Label rather than an Annotation, where a Label actually allows you to use a selector to select by that label.

Reply:
@xdevxy does the gitops-engine framework require the use of an Annotation or can we use a Label instead? (or we can use both if not)

Reply:
Oh, apologize that I should mention in the issue that we should utilize the cluster cache. Specifically, [LiveStateCache.GetManagedLiveObjs()](https://github.com/numaproj-labs/numaplane/blob/main/internal/sync/cache.go#L79rl) which you can get all the live objs associated to the given GitSync. I'll update the issue as well.